commons/src/config/locale.rs

```diff
@@ -67,6 +67,83 @@ pub mod serde {
                 .collect()
         }
     }
+
+    pub mod optional {
+        use super::*;
+
+        pub fn serialize<S>(locale: &Option<Locale>, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            match locale.as_ref().map(|l| l.to_string()) {
+                Some(locale) => serializer.serialize_some(&locale),
+                None => serializer.serialize_none(),
+            }
+        }
+
+        // The signature of a deserialize_with function must follow the pattern:
+        //
+        //    fn deserialize<'de, D>(D) -> Result<T, D::Error>
+        //    where
+        //        D: Deserializer<'de>
+        //
+        // although it may also be generic over the output types T.
+        pub fn deserialize<'de, D>(deserializer: D) -> Result<Option<Locale>, D::Error>
+        where
+            D: Deserializer<'de>,
+        {
+            let s = String::deserialize(deserializer)?;
+            match s.parse() {
+                Ok(locale) => Ok(Some(locale)),
+                Err(_) => Ok(None),
+            }
+        }
+    }
+
+    pub mod hashmap {
+        use super::*;
+        use std::collections::HashMap;
+        use std::path::PathBuf;
+
+        pub fn serialize<S>(locales_map: &HashMap<Locale, PathBuf>, serializer: S) -> Result<S::Ok, S::Error>
+            where
+                S: Serializer,
+        {
+            let mut res = String::new();
+
+            for entry in locales_map.iter() {
+                res.push_str("  ");
+                res.push_str(&entry.0.to_string());
+                res.push_str(": ");
+                res.push_str(&entry.1.to_str().expect("couldn't convert PathBuf to String"));
+                res.push('\n');
+            }
+
+            res.pop();
+
+            serializer.serialize_str(&res)
```

Comment:
```suggestion
            let mut map = serializer.serialize_map(Some(locales_map.len()))?;
            for (locale, path) in locales_map {
                map.serialize_key(&locale.to_string())?;
                map.serialize_value(path)?;
            }
            map.end()
```

Uses the map serializer from serde.